### PR TITLE
Changes to response selector documentation

### PR DESCRIPTION
**Pull request by dakshvar22**
**Proposed changes**:
- All changes to documentation because of https://github.com/RasaHQ/rasa/pull/6466

**Status (please check what you already did)**:
- [x] updated the documentation

### CODE REVIEW

File: docs/docs/components/selectors.mdx [hunk 1]
```diff
@@ -33,14 +34,35 @@ Selectors predict a bot response from a set of candidate responses.
   ```json
   {
       "response_selector": {
-        "faq": {
-          "response": {"confidence": 0.7356462617, "name": "Supports 3.5, 3.6 and 3.7, recommended version is 3.6"},
-          "ranking": [
-              {"confidence": 0.7356462617, "name": "Supports 3.5, 3.6 and 3.7, recommended version is 3.6"},
-              {"confidence": 0.2134543431, "name": "You can ask me about how to get started"}
+      "faq": {
```
Comment (erohmensing) [inline docs/docs/components/selectors.mdx:37]: please use a json prettifier on the output -- this should be indented

File: docs/docs/components/selectors.mdx [hunk 2]
```diff
@@ -52,6 +74,7 @@ Selectors predict a bot response from a set of candidate responses.
   a set of candidate responses. The prediction of this model is used by [Retrieval Actions](../retrieval-actions).
   It embeds user inputs and response labels into the same space and follows the exact same
   neural network architecture and optimization as the [DIETClassifier](../components/intent-classifiers.mdx#dietclassifier).
+  In its default configuration, it uses the retrieval intent with the response key(for e.g.- `faq/ask_name`) as the label for training.
```
Comment (erohmensing) [inline docs/docs/components/selectors.mdx:77]:
```suggestion
  In its default configuration, it uses the retrieval intent with the response key (e.g. `faq/ask_name`) as the label for training.
```

File: docs/docs/components/selectors.mdx [hunk 3]
```diff
@@ -110,10 +133,15 @@ Selectors predict a bot response from a set of candidate responses.
     set `weight_sparsity` to 1 as this would result in all kernel weights being 0, i.e. the model is not able
     to learn.
 
-  In addition, the component can also be configured to train a response selector for a particular retrieval intent.
-  The parameter `retrieval_intent` sets the name of the intent for which this response selector model is trained.
+  The component can also be configured to train a response selector for a particular retrieval intent.
+  The parameter `retrieval_intent` sets the name of the retrieval intent for which this response selector model is trained.
   Default is `None`, i.e. the model is trained for all retrieval intents.
 
+  In addition, the retrieval model can also be configured to use the text of the response messages as the training label
```
Comment (erohmensing) [inline docs/docs/components/selectors.mdx:140]:
```suggestion
  The retrieval model can also be configured to use the text of the response messages as the training label
```

Comment (erohmensing) [reply docs/docs/components/selectors.mdx:140]: Instead of what? maybe this section should go under where you talked about how it works by default

File: docs/docs/migration-guide.mdx [hunk 4]
```diff
@@ -383,7 +383,7 @@ Additionally a key `all_retrieval_intents` is added to the response selector out
 hold a list of all retrieval intents(faq, chitchat, etc.) that are present in the training data.
 An example output looks like this -
 
-```json
+```json {2,5,6}
 "response_selector": {
     "all_retrieval_intents": ["faq"],
     "default": {
```
Comment (erohmensing) [inline docs/docs/migration-guide.mdx:389]: Can we pretty this too? line 391 should have each of those on a separate line

File: docs/docs/migration-guide.mdx [hunk 5]
```diff
@@ -407,6 +407,17 @@ An example output looks like this -
         },
       ]
 ```
+As a result of this, if you were previously querying for the key `full_retrieval_intent` as -
+```python {2}
+response_selector_output.get("default")
+                        .get("full_retrieval_intent")
+```
+you should instead now do this -
```
Comment (erohmensing) [inline docs/docs/migration-guide.mdx:415]:
```suggestion
you should instead now do this:
```

File: docs/docs/migration-guide.mdx [hunk 6]
```diff
@@ -407,6 +407,17 @@ An example output looks like this -
         },
       ]
 ```
+As a result of this, if you were previously querying for the key `full_retrieval_intent` as -
```
Comment (erohmensing) [inline docs/docs/migration-guide.mdx:410]:
```suggestion
As a result of this, if you were previously querying for the key `full_retrieval_intent` as
```

File: docs/docs/retrieval-actions.mdx [hunk 7]
```diff
@@ -93,9 +93,10 @@ your responses.
 :::
 
 The retrieval model is trained separately as part of the NLU training pipeline
-to select the correct response. One important thing to remember is that the
-retrieval model uses the text of the response messages to select the correct
-one. If you change the text of these responses, you have to retrain your
+to select the correct response. The default configuration uses the user message text as input and the retrieval intent combined with the
+response key suffix(for e.g. - `chitchat/ask_name`) as the correct label for that user message. However, the
```
Comment (erohmensing) [inline docs/docs/retrieval-actions.mdx:97]:
```suggestion
response key suffix (e.g.`chitchat/ask_name`) as the correct label for that user message. However, the
```

File: docs/docs/retrieval-actions.mdx [hunk 8]
```diff
@@ -93,9 +93,10 @@ your responses.
...
+retrieval model can also be configured to use the text of the response message as the label for training the model.
```
Comment (erohmensing) [inline docs/docs/retrieval-actions.mdx:98]: how?

File: docs/docs/retrieval-actions.mdx [hunk 9]
```diff
@@ -127,11 +128,11 @@ pipeline:
 
 ## Domain
 
-Rasa uses a naming convention to match the intent names like `chitchat/ask_name`
-to the retrieval action.
+Rasa uses a naming convention to match the retrieval intent names `chitchat`
+to their corresponding retrieval action.
```
Comment (erohmensing) [inline docs/docs/retrieval-actions.mdx:132]:
```suggestion
Rasa uses a naming convention to match a retrieval intent name, e.g. `chitchat`,
to its corresponding retrieval action.
```
maybe?

File: docs/docs/retrieval-actions.mdx [hunk 10]
```diff
@@ -127,11 +128,11 @@ pipeline:
...
 The correct action name in this case is `respond_chitchat`.
 The prefix `respond_` is mandatory to identify it as a retrieval action.
-Another example - correct action name for `faq/ask_policy` would be `respond_faq`
+Another example - correct action name for `faq` would be `respond_faq`
```
Comment (erohmensing) [inline docs/docs/retrieval-actions.mdx:135]:
```suggestion
Another example - the correct action name for `faq` would be `respond_faq`.
```

File: docs/docs/retrieval-actions.mdx [hunk 11]
```diff
@@ -160,13 +161,27 @@ stories:
...
 
 If your assistant includes both FAQs **and** chitchat, it is possible to
-separate these into separate retrieval actions, for example having intents
-like `chitchat/ask_weather` and `faq/returns_policy`.
-Rasa supports adding multiple `RetrievalActions` like `respond_chitchat` and
-`respond_returns_policy`
+separate these into separate retrieval actions, for example having sub-intents
+like `chitchat/ask_weather` and `faq/returns_policy`. Rasa supports adding multiple `RetrievalActions`
+like `respond_chitchat` and `respond_faq`. To train separate retrieval models for each of the retrieval intents,
+you need to include a separate `ResponseSelector` component in the config:
```
Comment (erohmensing) [inline docs/docs/retrieval-actions.mdx:167]:
```suggestion
you need to include a separate `ResponseSelector` component in the config for each retrieval intent:
```

**Review by erohmensing — APPROVED** [review]
Thanks for going through all the different docs pages for this 💯

File: docs/docs/retrieval-actions.mdx (hunk 8 above)
Comment (dakshvar22) [reply docs/docs/retrieval-actions.mdx]: Added the name of the parameter from the component in the description.

File: docs/docs/components/selectors.mdx (hunk 3 above)
Comment (dakshvar22) [reply docs/docs/components/selectors.mdx]: Actually moved the default behaviour to this section as this section talks about the configuration of the model and hence more appropriate.
